### PR TITLE
17489 Partial Business Fetch component + clear button

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#17489

*Description of changes:*
- app version = 5.0.29
- implemented partial Business Fetch component
- deleted BusinessLookup from store
- used Business Fetch component when not logged in
- implemented business lookup/fetch clear button

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/components/new-request/business-fetch.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  /** When an item has been selected, emits event with business object. */
+  @Emit('business')
+  async onItemSelected (): Promise<BusinessLookupResultIF> {
+    const valid = this.$refs.searchField.validate()
+    if (!valid) return
+
+    // set state and perform search
+    this.state = States.SEARCHING
+    await Sleep(1000) // *** TODO: perform search here
```
Comment (severinbeauvais) [inline src/components/new-request/business-fetch.vue:69]: WIP but partly usable :)

File: src/components/new-request/business-fetch.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    // set state and return result
+    this.state = States.SUMMARY
+    return {
+      identifier: this.searchField,
+      legalType: CorpTypeCd.BC_COMPANY,
+      bn: '',
+      status: EntityStates.ACTIVE,
+      name: this.searchField
+    }
```
Comment (severinbeauvais) [inline src/components/new-request/business-fetch.vue:81]: For testing, using a business ID as search field.

File: src/components/new-request/search.vue [hunk 3]
```diff
@@ -96,10 +96,24 @@
...
+        <div v-else class="d-flex justify-space-between align-center">
+          <v-text-field
+            append-outer-icon="mdi-close"
+            disabled
+            filled
+            hide-details
+            :value="business.name"
+          />
+          <div @click="business=null">
+            <v-icon color="primary">mdi-close</v-icon>
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:114]: This is a usable clear/retry button, but I'm still waiting for Janis' design on it.

File: src/interfaces/new-request-interface.ts [hunk 4]
```diff
@@ -15,7 +15,6 @@ export interface NewRequestIF {
   analysisJSON: AnalysisJSONI
   applicant: ApplicantI
   assumedNameOriginal: string
-  businessLookup: BusinessLookupResultIF
```
Comment (severinbeauvais) [inline src/interfaces/new-request-interface.ts:18]: I think we don't need this in the store -- its lifespan is only while doing a business lookup on the Search page. After that, other variables are set from the data.

**Review by JazzarKarim — APPROVED** [review]
Looks great Sev!

**Review by ketaki-deodhar — APPROVED** [review]
LGTM!

**Comment by ketaki-deodhar** [comment]
Merging!

**Pull request merged**